process/coordinator/process_test.go

```diff
@@ -3017,5 +3018,1104 @@ func TestTransactionCoordinator_GetNumOfCrossShardSpecialTxsShouldWork(t *testin
+}
+
+func TestTransactionCoordinator_VerifyFeesShouldWork(t *testing.T) {
+	t.Parallel()
+
+	tx1GasLimit := uint64(100)
+
+	txHash := []byte("tx_hash1")
+	dataPool := initDataPool(txHash)
+	tc, err := NewTransactionCoordinator(
+		&mock.HasherMock{},
+		&mock.MarshalizerMock{},
+		mock.NewMultiShardsCoordinatorMock(3),
+		initAccountsMock(),
+		dataPool.MiniBlocks(),
+		&mock.RequestHandlerStub{},
+		createPreProcessorContainerWithDataPool(dataPool, FeeHandlerMock()),
+		createInterimProcessorContainer(),
+		&mock.GasHandlerMock{},
+		&mock.FeeAccumulatorStub{},
+		&mock.BlockSizeComputationStub{},
+		&mock.BalanceComputationStub{},
+		&mock.FeeHandlerStub{
+			DeveloperPercentageCalled: func() float64 {
+				return 0.1
+			},
+		},
+		&mock.TxTypeHandlerMock{},
+		0,
+	)
+	assert.Nil(t, err)
+	assert.NotNil(t, tc)
+
+	tx1 := &transaction.Transaction{Nonce: 1, GasLimit: tx1GasLimit, GasPrice: 1}
+	txHash1 := "hash1"
+
+	mapMiniBlockTypeAllTxs := make(map[block.Type]map[string]data.TransactionHandler)
+
+	mapAllTxs := make(map[string]data.TransactionHandler)
+	mapAllTxs[txHash1] = tx1
+	mapMiniBlockTypeAllTxs[block.TxBlock] = mapAllTxs
+
+	header := &block.Header{
+		AccumulatedFees: big.NewInt(100),
+		DeveloperFees:   big.NewInt(10),
+	}
+
+	body := &block.Body{
+		MiniBlocks: []*block.MiniBlock{
+			{
+				Type: block.PeerBlock,
+			},
+			{
+				Type:            block.TxBlock,
+				TxHashes:        [][]byte{[]byte(txHash1)},
+				ReceiverShardID: 1,
+			},
+		},
+	}
+
+	err = tc.verifyFees(header, body, mapMiniBlockTypeAllTxs)
+	assert.Nil(t, err)
+}
+
+func TestTransactionCoordinator_GetMaxAccumulatedAndDeveloperFeesShouldErr(t *testing.T) {
+	t.Parallel()
+
+	txHash := []byte("tx_hash1")
+	dataPool := initDataPool(txHash)
+	tc, err := NewTransactionCoordinator(
+		&mock.HasherMock{},
+		&mock.MarshalizerMock{},
+		mock.NewMultiShardsCoordinatorMock(3),
+		initAccountsMock(),
+		dataPool.MiniBlocks(),
+		&mock.RequestHandlerStub{},
+		createPreProcessorContainerWithDataPool(dataPool, FeeHandlerMock()),
+		createInterimProcessorContainer(),
+		&mock.GasHandlerMock{},
+		&mock.FeeAccumulatorStub{},
+		&mock.BlockSizeComputationStub{},
+		&mock.BalanceComputationStub{},
+		&mock.FeeHandlerStub{},
+		&mock.TxTypeHandlerMock{},
+		0,
+	)
+	assert.Nil(t, err)
+	assert.NotNil(t, tc)
+
+	txHash1 := "hash1"
+
+	mb := &block.MiniBlock{
+		Type:            block.TxBlock,
+		TxHashes:        [][]byte{[]byte(txHash1)},
+		ReceiverShardID: 1,
+	}
+
+	accumulatedFees, developerFees, errGetMaxFees := tc.getMaxAccumulatedAndDeveloperFees(mb, nil)
+	assert.Equal(t, process.ErrMissingTransaction, errGetMaxFees)
+	assert.Equal(t, big.NewInt(0), accumulatedFees)
+	assert.Equal(t, big.NewInt(0), developerFees)
+}
+
+func TestTransactionCoordinator_GetMaxAccumulatedAndDeveloperFeesShouldWork(t *testing.T) {
+	t.Parallel()
+
+	tx1GasLimit := uint64(100)
+	tx2GasLimit := uint64(200)
+	tx3GasLimit := uint64(300)
+
+	txHash := []byte("tx_hash1")
```

Comment:
`tx_hash1` is used 40 times in this file. maybe extract it to a const?

Reply:
Will do this in EN-8791 · Refactor TransactionCoordinator with arguments, at which I'm working it right now :)